src/SdkCommon/ClientRuntime/ClientRuntime/RetryDelegatingHandler.cs

```diff
@@ -117,15 +99,27 @@ public int EventCallbackCount
 
                     if (!responseMessage.IsSuccessStatusCode)
                     {
-                        // dispose the message unless we have stopped retrying
-                        this.Retrying += (sender, args) =>
```

Comment:
@AaronPatterson I see existing code has been deleted.
This PR cannot be accepted when existing functionality/code is being deleted.
Please add the feature without deleting any code or removing any existing functionality.

Reply:
What was being done previously was that a delegate was being registered each time a request was failed and before a retry happens the delegates would get invoked to allow them to do any cleanup they wanted to do.  The reason this was changed is that it was disposing of the previous response before it got a new one and this was one of the things which was leading to TaskCanceledExceptions instead of getting back the previously received error.  The updated code still does the disposing of the previous response, but it waits to do this until it has a new response.  With the new approach, the private Retrying delegate is no longer needed.

All of the externally observable functionality is still in place, so I do not believe that this is removing any functionality.